### PR TITLE
[utils] Variant: return matching iterators for (begin|end)_(array|map)

### PR DESCRIPTION
**Pull request by Rechi**
## Description
[utils] Variant: return matching iterators for (begin|end)_(array|map), if type is not array or map

With this change std::distance(...) is also correct if the type isn't an array or map.
Without this change there needs to be a check for isArray respectively isObject before iterating over the elements.

## Motivation and Context
Incorrect usage of CVariant at #12244.
@Montellese I don't know if you like this idea of the static array and map.

## How Has This Been Tested?
compiled and runtime tested on Windows and OSX

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/utils/Variant.cpp [hunk 1]
```diff
@@ -135,6 +135,8 @@ CVariant::CVariant()
 }
 
 CVariant CVariant::ConstNullVariant = CVariant::VariantTypeConstNull;
+CVariant::VariantArray CVariant::EMPTY_ARRAY = VariantArray();
+CVariant::VariantMap CVariant::EMPTY_MAP = VariantMap();
```
Comment (Montellese) [inline xbmc/utils/Variant.cpp]: You shouldn't need the `= VariantArray()` and `= VariantMap()` parts.

**Comment by Rechi** [comment]
@Montellese addressed your comment and runtime tested again.

**Review by Montellese — APPROVED** [review]
Good to go IMO.

**Comment by MartijnKaijser** [comment]
@Rechi could you make a backport PR?

**Comment by Rechi** [comment]
I don't think its necessary, but I'll backport it.

**Pull request merged**